### ISSUE TITLE
[Rule] html.security.audit.missing-integrity.missing-integrity doesn't work as expected

### ISSUE THREAD

**Issue by lfama** [returntocorp/sgrep-rules#2441]
**Describe the bug**

Rule [html.security.audit.missing-integrity.missing-integrity](https://semgrep.dev/r?q=html.security.audit.missing-integrity.missing-integrity) produces false positives with multiple consecutive `<link>` tags.

**To Reproduce**

The following code will trigger the rule even if it shouldn't:

```
<link rel="stylesheet" href="https://someurl/style.css" integrity="sha256-somehashdigest">
<link rel="stylesheet" href="./css/mystyle.css">
```

**Expected behavior**
The previous code snippet shouldn't trigger the rule, beacause the first link tag includes the `integrity` attribute and the second link tag is using a local `href` attribute (the rule is only looking for http(s) `href` attributes). 

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [X] P2: annoying but not blocking me

**Additional Context**

If you try to run the rule against each single link tag, it does not trigger (as it should be).

**Comment by LewisArdern** [comment]
Hey @lfama just made a PR which should solve this issue for you. Let me know if it causes any more problems.
